### PR TITLE
GPC header rule

### PR DESCRIPTION
**Pull request by jdorweiler**
Adds a modify header rule to add GPC to all requests. 
- https://privacy-test-pages.glitch.me/privacy-protections/gpc/

### CODE REVIEW

File: lib/rulePriorities.js [hunk 1]
```diff
@@ -6,3 +6,4 @@
 
 exports.AD_ATTRIBUTION_POLICY_PRIORITY = 20000
 exports.USER_ALLOWLISTED_PRIORITY = 1000000
+exports.GPC_HEADER_PRIORITY = 20000
```
Comment (kzar) [inline lib/rulePriorities.js:9]: This should be exported from the lib/gpc.js module instead.

File: test/gpc.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+                reason: '1st GPC header  reason'
+            }, {
+                domain: 'content-blocking2.example',
+                reason: '2nd GPC header  reason'
+            }]
+        }
+    }
+}
+
+const expectedGPCRule = [{
```
Comment (kzar) [inline test/gpc.js:26]: Maybe call it result, since it includes the lookup too?
```suggestion
const expectedGPCResult = [{
```

File: test/gpc.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+const {
+    generateGPCheaderRules
+} = require('../lib/gpc')
+
+const baseExtensionConfig = {
+    features: {
+        gpc: {
+            state: 'enabled',
+            exceptions: [{
+                domain: 'content-blocking1.example',
```
Comment (kzar) [inline test/gpc.js:16]: Maybe change the domain so it's clear it's unrelated to contentBlocking allowlist? (Same below.)
```suggestion
                domain: 'exception1.example',
```

File: test/gpc.js [hunk 4]
```diff
@@ -0,0 +1,62 @@
+const assert = require('assert')
```
Comment (kzar) [inline test/gpc.js:1]: Please could you add a test for the relative rule priorities of `GPC_HEADER_PRIORITY` in `test/rulePrioirities.js` too?

**Review by kzar — CHANGES_REQUESTED** [review]

File: lib/gpc.js [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    },
+    condition: { resourceTypes: [...resourceTypes] }
+}
+
+function generateGPCheaderRules (config) {
+    if (!(config.features.gpc && config.features.gpc.state === 'enabled')) {
+        return []
+    }
+
+    const rule = baseHeaderRule
```
[the remark this answers is not in the document]
Comment (jdorweiler) [reply lib/gpc.js]: I made a copy of the rule instead. `generateDNRRule` assume that you have always have `requestDomains` and fixing it was getting messy.

File: package.json [hunk 6]
```diff
@@ -21,6 +21,7 @@
         "eslint": "eslint . --ext .js",
         "tsc": "tsc",
         "lint": "npm run eslint && npm run tsc",
+        "eslint-fix": "npm run eslint -- --fix",
```
Comment (kzar) [inline package.json:24]: Nit: Mind putting this one below the "eslint" entry?

**Review by kzar — DISMISSED** [review]

**Review by kzar — APPROVED** [review]

**Pull request merged**